lib/sort_by_length.rb

```diff
@@ -1,7 +1,32 @@
 # A method which will return an array of the words in the string
 #  sorted by the length of the word.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n^2)
+# Space complexity: O(n)
+
 def sort_by_length(my_sentence)
-  raise NotImplementedError, "Method not implemented"
+  my_sentence = my_sentence.split(" ")                                  # Time: O(1), Space: O(n)
+  # implementing bubble sort
+  i = 0                                                                 # Time: O(1), Space: O(1)
+  while i < find_length(my_sentence) - 1                                # Time: O(n)
```

Comment:
calling `find_length` with every iteration of this loop makes it `O(n^4)`, since you do it in both loops